### PR TITLE
Update to use the canonical version of LB proto

### LINKED ISSUE #15728: Sync gRPC LB load_balancer.proto with Go and Java
In https://github.com/grpc/grpc-proto/pull/25 I added a canonical version of the load balancer proto that can be shared between all languages.  grpc code should sync with it so that all three grpc implementations can be aligned.  

cc: @ejona86

### PR DESCRIPTION
**Pull request by dgquintas**
For now, `load_balance.proto` is a copy of https://github.com/grpc/grpc-proto/blob/master/grpc/lb/v1/load_balancer.proto . In the future we may want to submodule `grpc/grpc-proto` instead.

The approach taken involves compiling `google.protobuf.duration` and `google.protobuf.timestamp` using nanopb, thus generating C code the core can compile and link. Most other changes are to auxiliary scripts used to perform this compilation. 

Fixes #15728

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                            FILE SIZE
 ++++++++++++++ GROWING                              ++++++++++++++
  +0.0%     +32 [None]                               +2.45Ki  +0.0%
      +0.0%     +32 [Unmapped]                           +2.45Ki  +0.0%
      [NEW]     +87 google_protobuf_Duration_fields          +87  [NEW]
      [NEW]     +87 google_protobuf_Timestamp_fields         +87  [NEW]

  +0.0%     +32 TOTAL                                +2.45Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                            FILE SIZE
 ++++++++++++++ GROWING                              ++++++++++++++
  +0.0%     +32 [None]                               +2.45Ki  +0.0%
      +0.0%     +32 [Unmapped]                           +2.45Ki  +0.0%
      [NEW]     +87 google_protobuf_Duration_fields          +87  [NEW]
      [NEW]     +87 google_protobuf_Timestamp_fields         +87  [NEW]

  +0.0%     +32 TOTAL                                +2.45Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                            FILE SIZE
 ++++++++++++++ GROWING                              ++++++++++++++
  +0.0%     +32 [None]                               +2.45Ki  +0.0%
      +0.0%     +32 [Unmapped]                           +2.45Ki  +0.0%
      [NEW]     +87 google_protobuf_Duration_fields          +87  [NEW]
      [NEW]     +87 google_protobuf_Timestamp_fields         +87  [NEW]

  +0.0%     +32 TOTAL                                +2.45Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                            FILE SIZE
 ++++++++++++++ GROWING                              ++++++++++++++
  +0.0%     +32 [None]                               +2.45Ki  +0.0%
      +0.0%     +32 [Unmapped]                           +2.45Ki  +0.0%
      [NEW]     +87 google_protobuf_Duration_fields          +87  [NEW]
      [NEW]     +87 google_protobuf_Timestamp_fields         +87  [NEW]

  +0.0%     +32 TOTAL                                +2.45Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/proto/grpc/lb/v1/load_balancer.proto [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016 gRPC authors.
+// Copyright 2015 The gRPC Authors
```
Comment (AspirinSJL) [inline src/proto/grpc/lb/v1/load_balancer.proto]: I'm just curious about why this gets decreased?

File: src/proto/grpc/lb/v1/load_balancer.proto [hunk 2]
```diff
@@ -12,42 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// This file defines an the GRPCLB LoadBalancing protocol.
```
Comment (AspirinSJL) [inline src/proto/grpc/lb/v1/load_balancer.proto]: Nit: "an" can be removed.

File: src/proto/grpc/lb/v1/load_balancer.proto [hunk 3]
```diff
@@ -62,10 +45,8 @@ message LoadBalanceRequest {
 }
 
 message InitialLoadBalanceRequest {
-  // The name of the load balanced service (e.g., balancer.service.com). The max
-  // length of the name is 256 bytes.
-  // The name might include a port number. How to handle the port number is up
-  // to the balancer.
+  // Name of load balanced service (IE, service.googleapis.com). Its
```
Comment (AspirinSJL) [inline src/proto/grpc/lb/v1/load_balancer.proto]: This overwrites my change before. I should probably update the canonical file.

File: src/proto/grpc/lb/v1/load_balancer.proto (hunk 1 above)
Comment (carl-mastrangelo) [reply src/proto/grpc/lb/v1/load_balancer.proto]: The earlier copyright takes precedence.  This file was copied a few times, but the original date holds.

File: tools/distrib/check_nanopb_output.sh [hunk 4]
```diff
@@ -41,21 +41,31 @@ make -j 8
 # back to the root directory
 popd
 
+##
+## Checks for load_balancer.proto
```
Comment (markdroth) [inline tools/distrib/check_nanopb_output.sh]: Why is this all commented out?

File: src/proto/grpc/lb/v1/load_balancer.proto (hunk 2 above)
Comment (dgquintas) [reply src/proto/grpc/lb/v1/load_balancer.proto]: @carl-mastrangelo can you fix that upstream?

File: tools/distrib/check_nanopb_output.sh (hunk 4 above)
Comment (dgquintas) [reply tools/distrib/check_nanopb_output.sh]: Fixed.

File: src/proto/grpc/lb/v1/load_balancer.proto (hunk 2 above)
Comment (carl-mastrangelo) [reply src/proto/grpc/lb/v1/load_balancer.proto]: @AspirinSJL actually has a PR upstream already, perhaps she can attach the change to her existing PR?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/proto/grpc/lb/v1/load_balancer.proto (hunk 2 above)
Comment (AspirinSJL) [reply src/proto/grpc/lb/v1/load_balancer.proto]: Oh, I thought this only exists in downstream copies. I will attach the change to my PR.

**Comment by AspirinSJL** [comment]
https://github.com/grpc/grpc-proto/pull/28 is merged. Please sync. Thanks!

**Comment by dgquintas** [comment]
Done. PTAL.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                            FILE SIZE
 ++++++++++++++ GROWING                              ++++++++++++++
  [ = ]       0 [None]                               +2.46Ki  +0.0%
      [ = ]       0 [Unmapped]                           +2.46Ki  +0.0%
      [NEW]     +87 google_protobuf_Duration_fields          +87  [NEW]
      [NEW]     +87 google_protobuf_Timestamp_fields         +87  [NEW]

  [ = ]       0 TOTAL                                +2.46Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by dgquintas** [comment]
Issues: #15827

**Pull request merged**